plugins/module_utils/openssh/certificate.py

```diff
@@ -339,21 +358,18 @@ class OpensshRSACertificateInfo(OpensshCertificateInfo):
     def __init__(self, e=None, n=None, **kwargs):
         super(OpensshRSACertificateInfo, self).__init__(**kwargs)
         self.type_string = _SSH_TYPE_STRINGS['rsa'] + _CERT_SUFFIX_V01
+        self.public_key_type_string = _SSH_TYPE_STRINGS['rsa']
         self.e = e
         self.n = n
 
     # See https://datatracker.ietf.org/doc/html/rfc4253#section-6.6
-    def public_key_fingerprint(self):
+    def write_public_key_params(self, writer):
         if any([self.e is None, self.n is None]):
-            return b''
+            return
```

Comment:
Returning early here without throwing an exception or any indication of missing data will result in an invalid fingerprint now instead of `b''`. So if you really want this refactor I would throw an exception and catch it in `public_key_fingerprint` to handle the error properly.

Reply:
Same goes for the other concrete implementations.